src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Formatters/StandardClassLibraryFormatter.cs

```diff
@@ -35,6 +40,581 @@ public byte[] Deserialize(ref MessagePackReader reader, MessagePackSerializerOpt
         }
     }
 
+    public sealed class SByteArrayFormatter : IMessagePackFormatter<sbyte[]>
+    {
+        public static readonly SByteArrayFormatter Instance = new SByteArrayFormatter();
+
+        private SByteArrayFormatter()
+        {
+        }
+
+        public unsafe void Serialize(ref MessagePackWriter writer, sbyte[] value, MessagePackSerializerOptions options)
+        {
+            if (value == null)
+            {
+                writer.WriteNil();
+                return;
+            }
+
+            var inputLength = value.Length;
+            writer.WriteArrayHeader(inputLength);
+            if (inputLength == 0)
+            {
+                return;
+            }
+
+            fixed (sbyte* pSource = &value[0])
+            {
+                var inputEnd = pSource + inputLength;
+                var inputIterator = pSource;
+
+#if HARDWARE_INTRINSICS_X86
+                const int Stride = 16;
+                if (Popcnt.IsSupported && inputLength >= Stride)
```

Comment:
Does this `if` check make the code safe to run in all contexts that offer the API? IOW does `Popcnt.IsSupported` safely return `false` on all non-supporting processors?

Reply:
[According to this official blog post](https://devblogs.microsoft.com/dotnet/hardware-intrinsics-in-net-core/#what-are-hardware-intrinsics), `Popcnt.IsSupported` property is trustable as far as .NET team implements this right.